lib/sycamore/sycamore/transforms/explode.py

```diff
@@ -28,17 +29,21 @@ def __init__(self, child: Node, **resource_args):
 
     @staticmethod
     @timetrace("explode")
-    def explode(parent: Document) -> list[Document]:
-        documents: list[Document] = [parent]
+    def explode(parent: Union[Document, HierarchicalDocument]) -> Union[list[Document], list[HierarchicalDocument]]:
+        if isinstance(parent, HierarchicalDocument):
+            return Explode.explode_hierarchical(parent)
+        if isinstance(parent, Document):
+            return Explode.explode_default(parent)
 
+    @staticmethod
+    @timetrace("explode")
+    def explode_default(parent: Document) -> list[Document]:
+        documents: list[Document] = [parent]
         import uuid
 
         for i, element in enumerate(parent.elements):
             cur = Document(element.data)
-            if "doc_id" in cur.data:
-                cur.doc_id = cur.data["doc_id"]
-            else:
-                cur.doc_id = str(uuid.uuid4())
+            cur.doc_id = str(uuid.uuid4())
```

Comment:
why did we change this?

Reply:
This was the previous implementation that I made. It was to make sure each entity in the entities property did not have their doc_id rewritten(since I had already built relationships between those id's and parent documents). We don't need this anymore since we switched over to using hierarchicaldocument class where the id's are assigned to children when they are converted to hierarchicaldocuments.